repos/shared-library/src/Framework/RenderMedia.php

```diff
@@ -18,6 +21,16 @@ public function render(
         $this->currentMedia = $media;
         $data = $media->mediaData();
 
+//        if (isset($data[AbstractIngester::SITE_ROLE_ID])) {
+//            /** @var User $user */
+//            $user = $media->getServiceLocator()->get('Omeka\AuthenticationService')->getIdentity();
+//            // Get the role from the site.
+//            $role = SitePermissionsHelper::userRoleForSite($user, $site);
+//            if (!in_array($role, $data[AbstractIngester::SITE_ROLE_ID])) {
+//                return '';
+//            }
+//        }
```

Comment:
This is something I need to rethink. There's no currently obvious way to get the current site when rendering media. I did have a working implementing that grabbed the route match, but that seems like needless overhead. Probably missing something, or something already in the DI container has had the site injected in and I've missed it.